src/rust/bitbox02-rust/src/hww/api/ethereum/sign.rs

```diff
@@ -299,6 +299,16 @@ pub async fn process(request: &Transaction<'_>) -> Result<Response, Error> {
     }
     super::keypath::warn_unusual_keypath(&params, params.name, request.keypath()).await?;
 
+    // Show chain confirmation only for known networks
+    if super::params::is_known_network(params.chain_id) {
```

Comment:
`let is_known_network = super::params::get(request.coin()?, request.chain_id()).is_some()` and delete the new helper function you had, as `get()` is backwards compatible with clients that use the `coin` parameter instead of the `chain_id` parameter.

Maybe it even makes sense to delete `get_and_warn_unknown()` and move the confirmation screen there to here, so they are all close together, or re-factor it into `fn warn_unknown()` and call it here. The warning also appears when displaying a receive address, but I am not sure anymore if this is needed there at all, as the address is the same everywhere anyway.

Reply:
ok, refactored as you suggested (I think, lmk if I misunderstood what you meant)
see the fixup commit for the diff

Reply:
Sorry for the trouble, but let's revert it :sweat_smile:  The reasons are:

- using `::get()` instead of `::get_and_warn_unknown()` in the pubrequest means the call will fail for unknown networks, which is not desirable. We should think about if we want to remove the coin/chain_id param from that call completely, but that should be a separate discussion/PR.
- warn_unknown does not only warn, it also produces params, and the prompts are still not closer together than they were before :shrug: 

I did not foresee these before, sorry.

Let's revert these parts, but make `is_known_network(coin, chain_id)` simply call `params::get(coin, chain_id).is_some()`.

Reply:
> warn_unknown does not only warn, it also produces params, and the prompts are still not closer together than they were before 

hmm, yeah, I see that now.

> Let's revert these parts, but make is_known_network(coin, chain_id) simply call params::get(coin, chain_id).is_some()

Then keep the change making the getter public right, to call it from sign.rs?